python/cudf/cudf/tests/test_reshape.py

```diff
@@ -137,12 +137,19 @@ def test_df_stack(nulls, num_cols, num_rows, dtype):
 def test_interleave_columns(nulls, num_cols, num_rows, dtype):
     if dtype not in ["float32", "float64"] and nulls in ["some"]:
         pytest.skip(msg="nulls not supported in dtype: " + dtype)
+    
+    broken_shapes = [(1,1), (1,2), (1,10), 
```

Comment:
@kkraus14 So I'm getting some odd behavior with `category` tests and some shapes. Not having the most robust understanding of the `category` dtype, I'm not sure how to label the problem. It could also have to do with my pandas interleave method below on L165. I skipped the complaining DF shapes for `category` in the test so I could identify the issue explicitly, but I would much rather fix the problem.

Here are the errors I'm getting.

```
E       AssertionError: Attributes are different
E       
E       Attribute "dtype" are different
E       [left]:  CategoricalDtype(categories=[1, 4, 5, 7, 8, 11, 12, 13, 14, 15, 17, 18, 21, 24], ordered=False)
E       [right]: CategoricalDtype(categories=[17, 18], ordered=False)

cudf/tests/test_reshape.py:175: AssertionError
```
```
E           ValueError: codes need to be between -1 and len(categories)-1

../../../compose/etc/conda/envs/rapids/lib/python3.7/site-packages/pandas/core/arrays/categorical.py:707: ValueError
```
Potentially it's related to the copying of categories but I'm scratching my head to figure out why some DF shapes work fine.

Reply:
Hey @millerhooks , I had to bang my head against this same categorical issue when reworking some pieces of `join` a few weeks ago and I think I understand it fairly well at this point. Feel free to DM me if you want to chat, I might be able to shed some light on this.

Reply:
Awesome! Thanks Brandon!